File: PA1_Server/src/edu/ucdenver/tournament/Player.java
package edu.ucdenver.tournament;

import java.io.Serializable;

/**
 * <p>This defines the Player class.
 * The Player class is responsible
 * for creating a player and returning
 * necessary information.
 * </p>
 * @author Karim Kabbara
 * @version 1.0
 * @since 2022-11-11
 */
public class Player implements Serializable {

    /**
     * <p>
     *     Player's name
     * </p>
     */
    private final String name;

    /**
     * <p>
     *     Player's age
     * </p>
     */
    private final int age;

    /**
     * <p>
     *     Player's height
     * </p>
     */
    private final double height;

    /**
     * <p>
     *     Player's weight
     * </p>
     */
    private final double weight;

    /**
     * <p>
     * This is the constructor
     * for the player class. It sets
     * the name, age, height and weight.
     * </p>
     * @param name          Player Name
     * @param age           Player Age
     * @param height        Player Height
     * @param weight        Player Weight
     */
    public Player(String name, int age, double height, double weight) {
        this.name = name;
        this.age = age;
        this.height = height;
        this.weight = weight;
    }

    /**
     * <p>
     * Returns player name
     * </p>
     * @return Player name
     */
    public String getName() {
        return this.name;
    }

    /**
     * <p>
     * Returns player age
     * </p>
     * @return Player age
     */
    public int getAge() {
        return this.age;
    }

    /**
     * <p>
     * Returns player height
     * </p>
     * @return Player height
     */
    public double getHeight() {
        return this.height;
    }

    /**
     * <p>
     * Returns player weight
     * </p>
     * @return Player weight
     */
    public double getWeight() {
        return this.weight;
    }
}

File: Server (Python)/src/Player.py
class Player:
    """
    This defines the Player class.
    The Player class is responsible
    for creating a player and returning
    necessary information.
    """
    _name: str  # player name
    _age: int   # player age
    _height: float  # player height
    _weight: float  # player weight

    def __init__(self, name: str, age: int, height: float, weight: float):
        """
        This is the constructor
        for the player class. It sets
        the name, age, height and weight.
        :param name: player name
        :param age: player age
        :param height: player height
        :param weight: player weight
        """
        self._name = name
        self._age = age
        self._height = height
        self._weight = weight

    @property
    def get_name(self) -> str:
        """
        Returns player name
        :return: player name (String)
        """
        return self._name

    @property
    def get_age(self) -> int:
        """
        Gets player age
        :return: player age (Int)
        """
        return self._age

    @property
    def get_height(self) -> float:
        """
        Gets player height
        :return: player height (Float)
        """
        return self._height

    @property
    def get_weight(self) -> float:
        """
        Returns player weight
        :return: player weight (Float)
        """
        return self._weight
